tmail-backend/mailbox/encrypted/api/src/main/scala/com/linagora/tmail/encrypted/EncryptedEmailContent.scala

```diff
@@ -0,0 +1,82 @@
+package com.linagora.tmail.encrypted
+
+import com.google.common.io.ByteSource
+import com.linagora.tmail.pgp.Encrypter
+import org.apache.james.jmap.api.model.Preview
+import org.apache.james.mailbox.model.{MessageId, ParsedAttachment}
+
+import java.io.ByteArrayOutputStream
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.atomic.AtomicInteger
+import scala.jdk.OptionConverters._
+
+class EncryptedEmailContentFactory(encrypter: Encrypter) {
+
+  def encrypt(clearEmailContent: ClearEmailContent, messageId: MessageId): EncryptedEmailContent =
+    EncryptedEmailContent(encryptedPreview = encryptPreview(clearEmailContent.preview),
+      encryptedHtml = encrypt(clearEmailContent.html),
+      hasAttachment = clearEmailContent.hasAttachment,
+      encryptedAttachmentMetadata = encryptAttachmentMetadata(clearEmailContent.attachments, messageId),
+      encryptedAttachmentContents = encryptAttachmentContent(clearEmailContent.attachments))
+
+  def encryptPreview(preview: Preview): String =
+    encrypt(preview.getValue)
+
+  def encryptAttachmentMetadata(parsedAttachments: List[ParsedAttachment], messageId: MessageId): Option[String] = {
+    if (parsedAttachments.isEmpty) return None
+    Some(encrypt(AttachmentMetaDataSerializer
+      .serializeList(AttachmentMetadata.fromJava(parsedAttachments, messageId))
```

Comment:
To what I said previously I would then create at the beginning of this method an atomic integer, that you start at -1, then you pass each time here the integer and you increment it same time (like using `incrementAndGet()` directly as an argument).

Then it allows you to remove your method `def fromJava(listParsedAttachment: List[ParsedAttachment], messageId: MessageId)` as you would not need it anymore